docs: `rover init` (#2451)
apollographql/apollo-cli: docs/source/commands/init.mdx

---
title: The Rover init Command
subtitle: Initialize a GraphQL API project
description: Use the Rover CLI init command to initialize a federated GraphQL API project using Apollo Federation and the GraphOS Router.
minVersion: 0.29.0
---

> `rover init` is an interactive command designed to help you create a new federated graph locally. It generates all the necessary configuration files and boilerplate code required to get started.

Reviewer: Should we also mention that it will sync the graph to GraphOS?

Reviewer: Or maybe we remove 'locally'?

Author: Good point! I like the clarity of mentioning that the command creates a local graph  _and_ pushes it to GraphOS.

Author: Edit: I ended up removing locally in the intro to simplify things and then made it clearer later on. https://github.com/apollographql/rover/pull/2451/commits/05d143654b90524770a0a41baf262f052fb9c71d

## Initializing a graph

Running `rover init` starts a short wizard that helps you create a new graph.

```terminal showLineNumbers=false
rover init
```

The wizard walks you through a set of questions to help you choose the best option for your use case. By the end, it generates:

- A set of local files, including a `README.md` and `getting-started.md`

> - The necessary credentials to interact with the GraphOS platform

Reviewer: also connects the graph to GraphOS

### Choosing your use case

Once you run `rover init`, the wizard prompts you to select your use case.

> - **Start a GraphQL API with one or more REST APIs**: Select this option to use Apollo Connectors to quickly integrate REST APIs into your graph.
> - [Coming Soon] **Start a GraphQL API with recommended libraries**: Select this option if you want to connect data that's not accessible via a REST API. This option helps you set up a GraphQL API using the Apollo Server library.

Reviewer: We are updating copy again to change `GraphQL API` to `graph`

```suggestion
- **Start a graph with one or more REST APIs**: Select this option to use Apollo Connectors to quickly integrate REST APIs into your graph.
- [Coming Soon] **Start a graph with recommended libraries**: Select this option if you want to connect data that's not accessible via a REST API. This option helps you set up a GraphQL API using the Apollo Server library.
```

Author: https://github.com/apollographql/rover/pull/2451/commits/4031c5c3702d5e578b0ea7aae4e233b463659860

---
title: The Rover init Command
subtitle: Initialize a GraphQL API project
description: Use the Rover CLI init command to initialize a federated GraphQL API project using Apollo Federation and the GraphOS Router.
minVersion: 0.29.0
---

`rover init` is an interactive command designed to help you create a new federated graph locally. It generates all the necessary configuration files and boilerplate code required to get started.

## Initializing a graph

> Running `rover init` starts a short wizard that helps you create your new GraphQL API project.

Author: ```suggestion
Running `rover init` starts a short wizard that helps you create a new graph.
```

```terminal showLineNumbers=false
rover init
```

The wizard walks you through a set of questions to help you choose the best option for your use case. By the end, it generates:

- A set of local files, including a README.md
- The necessary credentials to interact with the GraphOS platform

### Choosing your use case

Once you run `rover init`, the wizard prompts you to select your use case.

> - **Connect one or more REST APIs**: Select this option to use Apollo Connectors to quickly integrate REST APIs into your graph.

Reviewer: This will need to be updated based on the changes to design
```suggestion
- **Start a GraphQL API with one or more REST APIs**: Select this option to use Apollo Connectors to quickly integrate REST APIs into your graph.
```

Author: https://github.com/apollographql/rover/pull/2451/commits/96f250e139150c9670e8f702829b87ee00e4d665

- [Coming Soon] **Start a GraphQL API with recommended libraries**: Select this option if you want to connect data that's not accessible via a REST API. This option helps you set up a GraphQL API using the Apollo Server library.

### Created credentials

After you've chosen your use case, the wizard prompts you for a project name. It generates the following credentials based on the name you enter:

- **Graph ID**- This is a unique identifier for your graph. It represents your graph across all of Apollo.
- **Graph ref**- A graph ref references a specific variant of your graph. It's formatted `graph-id@variant`. To start, GraphOS automatically creates a variant titled `current` for you.
- **Graph API key**- This is your graph's API key. Once it's generated, store it securely—you won't be able to access it later.

### Using Apollo's IDE extensions

> Finally, the wizard asks if you would like your project to include settings for one of Apollo's IDE extensions. See the dedicated documentation pages linked on the [IDE extensions overview](/graphos/schema-design/ide-support) for further configuration instructions.

Author: I recall there was some discussion about how `rover init` would work with IDE settings. Is this still accurate?

Reviewer: We are no longer prompting the user to choose their IDE and instead have opted to move this to `getting-started.md`.

Author: https://github.com/apollographql/rover/pull/2451/commits/ff1705e4524b676fbbbc1476a03e3e22c0b50ee5

---
title: The Rover init Command
subtitle: Initialize a GraphQL API project
description: Use the Rover CLI init command to initialize a federated GraphQL API project using Apollo Federation and the GraphOS Router.
minVersion: 0.29.0
---

`rover init` is an interactive command designed to help you create a new federated graph locally. It generates all the necessary configuration files and boilerplate code required to get started.

## Initializing a graph

Running `rover init` starts a short wizard that helps you create your new GraphQL API project.

```terminal showLineNumbers=false
rover init
```

The wizard walks you through a set of questions to help you choose the best option for your use case. By the end, it generates:

- a set of local files, including a README.md

> - the necessary credentials to interact with the GraphOS platform

Author: ```suggestion
- A set of local files, including a README.md
- The necessary credentials to interact with the GraphOS platform
```

### Choosing your use case

Once you run `rover init`, the wizard prompts you to select your use case.

- **Connect one or more REST APIs**: Select this option to use Apollo Connectors to quickly integrate REST APIs into your graph.

> - **Start a GraphQL API with recommended libraries**: Select this option if you want to connect data that's not accessible via a REST API. This option helps you set up a GraphQL API using the Apollo Server library.

Author: ```suggestion
- [Coming Soon] **Start a GraphQL API with recommended libraries**: Select this option if you want to connect data that's not accessible via a REST API. This option helps you set up a GraphQL API using the Apollo Server library.
```

- **Graph ID**- This is a unique identifier for your graph. It represents your graph across all of Apollo.
- **Graph ref**- A graph ref references a specific variant of your graph. It's formatted `graph-id@variant`. To start, GraphOS automatically creates a variant titled `current` for you.
- **Graph API key**- This is your graph's API key. Once it's generated, store it securely—you won't be able to access it later.

### Using Apollo's IDE extensions

Finally, the wizard asks if you would like your project to include settings for one of Apollo's IDE extensions. See the dedicated documentation pages linked on the [IDE extensions overview](/graphos/schema-design/ide-support) for further configuration instructions.

## Additional resources

- If you chosen to connect one or more REST APIs, refer to the [Apollo Connectors getting started guide](/graphos/get-started/guides/rest) for next steps.
- If you've chosen to start a GraphQL API using Apollo Server, refer to:
    - the [GraphQL getting started guide](/graphos/get-started/guides/graphql) for an introduction based on Apollo's retail demo graph.

>     - the [`rover template` command](/rover/commands/template) for additional starting points.

Author: ```suggestion
## Next steps

- Check out the [Getting started guide](/graphos/get-started/guides/rest) to learn what else you can do with Apollo Connectors.
- If you learn best with videos and exercises, this [interactive course](https://www.apollographql.com/tutorials/connectors-intro-rest) teaches you how to bring an existing REST API into a GraphQL API using Apollo Connectors.
```
